### PR TITLE
ddl: fix parallel create table if not exists

### PR DESCRIPTION
**Pull request by winkyao**
In onCreateTable, we will check if table is exists, but we use the `if not exists` option only in CreateTable. It will cause unexpected ` [schema:1050]Table 'Xxx' already exists`

### CODE REVIEW

File: ddl/ddl_db_change_test.go [hunk 1]
```diff
@@ -433,7 +433,6 @@ func (s *testStateChangeSuite) TestShowIndex(c *C) {
 }
 
 func (s *testStateChangeSuite) TestParallelDDL(c *C) {
-	defer testleak.AfterTest(c)()
```
Comment (shenli) [inline ddl/ddl_db_change_test.go]: Why remove this?

File: ddl/ddl_db_change_test.go [hunk 2]
```diff
@@ -501,3 +499,85 @@ func (s *testStateChangeSuite) TestParallelDDL(c *C) {
 	callback = &ddl.TestDDLCallback{}
 	d.SetHook(callback)
 }
+
+func (s *testStateChangeSuite) TestCreateIfNotExists(c *C) {
```
Comment (shenli) [inline ddl/ddl_db_change_test.go]: Add some comments for the test case.

File: ddl/ddl_db_change_test.go [hunk 3]
```diff
@@ -501,3 +499,85 @@ func (s *testStateChangeSuite) TestParallelDDL(c *C) {
...
+		_, err3 = se1.Execute(context.Background(), "create table if not exists test_not_exists(a int);")
+	}()
+	wg.Wait()
+	c.Assert(err2, IsNil)
+	c.Assert(err3, IsNil)
+	callback = &ddl.TestDDLCallback{}
+	d.SetHook(callback)
+}
+
+func (s *testStateChangeSuite) TestCreateIfNotExists1(c *C) {
```
Comment (shenli) [inline ddl/ddl_db_change_test.go]: ditto

File: ddl/ddl_db_change_test.go (hunk 1 above)
Comment (winkyao) [reply ddl/ddl_db_change_test.go]: testleak is verified in testStateChangeSuite.TeareDownSuite, call `testleak.AfterTest(c)` here make test unstable.

**Comment by winkyao** [comment]
@zimulala PTAL

File: ddl/ddl_db_change_test.go [hunk 4]
```diff
@@ -501,3 +499,56 @@ func (s *testStateChangeSuite) TestParallelDDL(c *C) {
...
+	_, err = se1.Execute(context.Background(), "use test_db_state")
+	c.Assert(err1, IsNil)
+
+	var err2, err3 error
+	wg := sync.WaitGroup{}
+
+	callback := &ddl.TestDDLCallback{}
+	once := sync.Once{}
+	callback.OnJobUpdatedExported = func(job *model.Job) {
+		// sleep a while, let other job enque.
```
Comment (zimulala) [inline ddl/ddl_db_change_test.go]: How can we confirm that the other job is in the queue?

File: ddl/ddl_api.go [hunk 5]
```diff
@@ -856,6 +856,11 @@ func (d *ddl) CreateTable(ctx sessionctx.Context, s *ast.CreateTableStmt) (err e
 			err = d.handleAutoIncID(tbInfo, schema.ID)
 		}
 	}
+
+	// table exists, but if_not_exits flags is true, so we ignore this error.
```
Comment (lamxTyler) [inline ddl/ddl_api.go]: `if_not_exists`

File: ddl/ddl_db_change_test.go (hunk 4 above)
Comment (winkyao) [reply ddl/ddl_db_change_test.go]: Before this PR, the added test failed stably.

Comment (zimulala) [reply ddl/ddl_db_change_test.go]: But this test code is indeed unstable.

**Comment by winkyao** [comment]
@zimulala @lamxTyler  PTAL

**Pull request merged**